src/Http/RequestFactory.php

```diff
@@ -112,19 +112,20 @@ public function createHttpRequest()
 		}
 
 		// remove invalid characters
+		$reChars = '#^[' . self::CHARS . ']*+\z#u';
 		if (!$this->binary) {
 			$list = array(& $query, & $post, & $cookies);
 			while (list($key, $val) = each($list)) {
 				foreach ($val as $k => $v) {
-					if (is_string($k) && (!preg_match(self::CHARS, $k) || preg_last_error())) {
+					if (is_string($k) && (!preg_match($reChars, $k) || preg_last_error())) {
 						unset($list[$key][$k]);
 
 					} elseif (is_array($v)) {
 						$list[$key][$k] = $v;
 						$list[] = & $list[$key][$k];
 
-					} elseif (!preg_match(self::CHARS, $v) || preg_last_error()) {
-						$list[$key][$k] = '';
+					} else {
+						$list[$key][$k] = (string) preg_replace('#[^' . self::CHARS . ']+#u', '', $v);
```

Comment:
Have you compared performance with doing preg_match (to check if replace is needed) first?